hail/python/test/hailtop/utils/test_gcs_requester_pays.py

```diff
@@ -0,0 +1,27 @@
+from operator import attrgetter
+from pytest import raises
+
+from hailtop.aiotools.router_fs import RouterAsyncFS
+from hailtop.fs.router_fs import RouterFS
+from hailtop.utils import async_to_blocking
+from hailtop.utils.gcs_requester_pays import gcs_requester_pays_fs_cache
+
+
+def test_get_fs_by_requester_pays_config():
+    def test_with_cls(cls, kwargs_field):
+        config_1 = "foo"
+        config_2 = ("foo", ["bar", "baz", "bat"])
+        kwargs_1 = {"gcs_requester_pays_configuration": config_1}
+        kwargs_2 = {"gcs_requester_pays_configuration": config_2}
+        fses = gcs_requester_pays_fs_cache(cls)
+        assert attrgetter(kwargs_field)(fses[None]) == {}
+        assert attrgetter(kwargs_field)(fses[config_1]) == kwargs_1
+        set_kwargs = attrgetter(kwargs_field)(fses[config_2])
+        assert set_kwargs["gcs_requester_pays_configuration"][0] == config_2[0]
+        assert len(set_kwargs["gcs_requester_pays_configuration"][1]) == len(config_2[1])
+        assert set(set_kwargs["gcs_requester_pays_configuration"][1]) == set(config_2[1])
+        default_kwargs_fses = gcs_requester_pays_fs_cache(cls, {"gcs_kwargs": kwargs_1})
+        assert attrgetter(kwargs_field)(default_kwargs_fses[None]) == kwargs_1
+
+    test_with_cls(RouterFS, "afs._gcs_kwargs")
+    test_with_cls(RouterAsyncFS, "_gcs_kwargs")
```

Comment:
I feel like we can instead make `test_with_cls` the actual test method and use `pytest.mark.parametrize` with the two classes.

Reply:
updated!